src/bucklescript/output_bucklescript_types.re

```diff
@@ -409,11 +407,8 @@ let generate_arg_types = (config, variable_defs) => {
   let input_objects = extract_args(config, variable_defs);
 
   input_objects
-  |> List.map(
-       fun
-       | InputObject({name, fields}) => {
```

Comment:
I think these are premature refactorings. Because the type only has a single variant now doesn't mean that it stays that way. In the old syntax there is clearly one variant being handled in a switch statement making it easy to extend later. And I am not sure if we lost exhaustiveness checking with this refactor.

Reply:
I'm not sure about default dune defaults but there should be an exhaustiveness check.
I did it because I think the amount of indentation makes things harder to read. But probably it's not the best idea and we should focus on extracting smaller functions rather than doing such premature optimizations.

Reply:
I'm not sure about default dune defaults but there should be an exhaustiveness check.
I did it because I think the amount of indentation makes things harder to read. But probably it's not the best idea and we should focus on extracting smaller functions rather than doing such premature optimizations.

Reply:
If exhaustiveness checking still works it's fine. And changing it later is also not a lot of work.

Reply:
If there are not extra cases we can remove the variant type, I started with a variant because I thought there would be more cases, but maybe I was wrong here.